vortex-file/src/write/stats_accumulator.rs

```diff
@@ -86,9 +66,12 @@ impl StatsAccumulator {
             return Ok(None);
         }
 
-        Ok(Some(
+        Ok(Some(StatArray(
             StructArray::try_new(names.into(), fields, self.length, Validity::NonNullable)?
                 .into_array(),
-        ))
+            stats,
+        )))
     }
 }
+
+pub struct StatArray(pub ArrayData, pub Vec<Stat>);
```

Comment:
not sure this is better than anonymous tuple

Reply:
No it's not, but `struct StatisticsArray { array: ArrayData, present_stats: Vec<Stat> }` is!

Then we can `impl PruningStatistics for StatisticsArray` and lots of other things get cleaner.

Reply:
I don't think I follow your suggestion, `PruningStatistics` is not a thing in our codebase